python/python/bystro/supervised_ppca/ppca_analytic_np.py

```diff
@@ -0,0 +1,101 @@
+"""
+This implements the standard version of probabilistic PCA found in Chapter
+12 of Bishop (2006). It has an analytic solution making this method very
+quick to fit. Will serve as a baseline comparison for many of our methods.
+
+Objects
+-------
+PPCAanalytic(n_components=2)
+    The plain vanilla probabilistic PCA model with an analytic formula that
+    is estimated via maximum likelihood.
+
+Methods
+-------
+None
+"""
+import numpy as np
+import numpy.linalg as la
+from bystro.supervised_ppca._base import BaseGaussianFactorModel
+
+
+class PPCAanalytic(BaseGaussianFactorModel):
+    def __init__(self, n_components=2):
+        """
+        Analytic PPCA solution as described by Bishop
+
+        Parameters
+        ----------
+        n_components : int,default=2
+            The latent dimensionality
+        """
+        super().__init__(n_components=n_components)
+
+    def __repr__(self):
```

Comment:
This is mostly a surprise because Python’s defaults tend to be fairly useful. However, in this case, having a default for __repr__ which would act like:

return "%s(%r)" % (self.__class__, self.__dict__)
would have been too dangerous (for example, too easy to get into infinite recursion if objects reference each other). So Python cops out. Note that there is one default which is true: if __repr__ is defined, and __str__ is not, the object will behave as though __str__=__repr__.

This means, in simple terms: almost every object you implement should have a functional __repr__ that’s usable for understanding the object. Implementing __str__ is optional: do that if you need a “pretty print” functionality (for example, used by a report generator).

I'm really lazy and would prefer not to make two methods. But willing to be overruled.

Reply:
I'm lazy and would rather just implement a single method.

Reply:
ok, for consistency with sklearn I would recommend just returning `f'PPCAAnalytic(n_components={self.n_components})'` then.  Sklearn objects return a string formatted in that style for both str and repr

Reply:
OK sounds good.